base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigure.java

```diff
@@ -154,10 +179,23 @@ public void mouseUp(MouseEvent e) {
 					    plotMessage.setText(e.getNewValue().toString());
 					}
 				});
-		imageListener = viewModel.getImage()
-				.addUiThreadPropertyChangeListener(e -> drawImage((ImageData) e.getNewValue()));
-		
-		
+		cursorTypeListener = viewModel.getCursorType()
+				.addUiThreadPropertyChangeListener(e -> {
+					if (!plotMessage.isDisposed()) {
+					    if (viewModel.getCursorType().getValue() == MatplotlibCursorType.CROSSHAIR) {
```

Comment:
As general good-practice in java, we should use `MatplotlibCursorType.CROSSHAIR.equals(viewModel.getCursorType().getValue())` - `==` is checking reference equality, not object equality. I think we get away with it here regardless but best to change it anyway...